include/mata/utils/ord-vector.hh

```diff
@@ -146,6 +146,13 @@ public:
         vec_.emplace_back(x);
     }
 
+    //I added this, and then noticed that that push_back above does emplace_back already.
+    // Isn't that somehow wrong? Is emplace_back really always better?
+    virtual inline void emplace_back(const Key& x) {
```

Comment:
There is a difference between them. See a [nice article](https://andreasfertig.blog/2023/04/push_back-vs-emplace_back-when-to-use-what/) about when to use `emplace_back()` or have a look at the [example](https://en.cppreference.com/w/cpp/container/vector/emplace_back#Example) in `emplace_back()` at C++ Reference.

We can have both, or just one. But if the function signatures are always with `const Key&`, both will run the same code: construct a temporary value of type `Key` and move it into the vector (the behaviour of `push_back()`.

Reply:
We do not always use primitive data types, e.g. vectors of symbol_pos. There I could take advantage of emplace_back, no?

Reply:
Yes, for cases where we have composite data types, such as `SymbolPost`, we can utilize `emplace_back()` very efficiently.

Reply:
Now the problem may be that people were already relying on push_back actually doing emplace_back. If we change it, we have to go through all calls of push_back. Annoying.

Reply:
And in fact I still don't get it, it looks lika doing it the current way, the push_back calls emplace_back, has only advantages, since if emplace_back makes sense, it will work faster than push_back, and if it does not make sense, it will do the same thing as push_back. Or what am I missing? In other words, why would anyone ever want to call vector::push_back actually? Just call emplace_back always, no?

Reply:
Resolved in person. Additional implementation of `emplace_back()` which allows in-place constructions added.